src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -1076,12 +1076,14 @@ public static Set<PIP> getTrimmablePIPsFromPins(Net net, Collection<SitePinInst>
 	            rPips.add(pip);
 	        }
 
-	        Integer count = fanout.get(startNode);
-	        if(count == null){
-	            fanout.put(startNode, 1);
-	        }else{
-	            fanout.put(startNode, count+1);
-	        }
+	        SitePin sp = startNode.getSitePin();
+	        SiteInst si = (sp != null && sp.isInput()) ? net.getDesign().getSiteInstFromSite(sp.getSite()) : null;
```

Comment:
A few comments:

1) It might be best to store a reference to the `Design` object outside of the PIP loop.
2) Should we guard the last two lines with a null check for `sp`?  It would reduce the number of null checks for the common case.
3) Does this not happen for the `endNode` case?

Reply:
For 2. which are the last two lines you are referring to? Feel free to use the code suggestion, I often find it easier/clearer to write small snippet of code to show what I mean than describe it using English. Pfffft to that imprecise language!

Re: 3. My thinking was that we only care about incrementing fanout if it is both a node-with-site-pin and is used by a PIP. You can't detect that condition by looking at the `endNode` of the PIP that takes you to the input site pin, but you can detect the condition on the following PIP that takes you away from the input pin. Is that logic sound?
In any case, this loop is about computing the fanout count of the `startNode`.